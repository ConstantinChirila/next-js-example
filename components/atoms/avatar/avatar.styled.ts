import styled from 'styled-components'

export const StyledAvatar = styled.img`
  border-radius: ${({ theme }) => theme.radius.lg};
  display: block;
  height: 6.4rem;
  object-fit: cover;
  object-position: center;
  width: 6.4rem;
`
