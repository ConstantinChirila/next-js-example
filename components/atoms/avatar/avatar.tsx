import React, { ReactElement } from 'react'

import { StyledAvatar } from './avatar.styled'
import { TAvatarProperties } from './avatar.types'

export function Avatar(properties: TAvatarProperties): ReactElement {
  return <StyledAvatar {...properties} />
}
