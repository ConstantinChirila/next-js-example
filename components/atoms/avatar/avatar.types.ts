import { ImgHTMLAttributes } from 'react'

export type TAvatarProperties = ImgHTMLAttributes<HTMLImageElement>
