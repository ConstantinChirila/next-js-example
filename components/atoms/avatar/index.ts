export * from './avatar'
export * from './avatar.styled'
export * from './avatar.types'
