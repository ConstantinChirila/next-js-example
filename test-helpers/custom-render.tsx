import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
