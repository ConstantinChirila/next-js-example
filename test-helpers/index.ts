export * from './custom-render'
export * from './setup'
