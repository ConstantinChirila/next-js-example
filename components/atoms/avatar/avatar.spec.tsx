import React from 'react'

import { Avatar } from './avatar'
import { renderWithTheme } from '../../../test-helpers'

describe('Avatar', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(
      <Avatar alt="hello alt" src="http://placekitten.com/90/90" />
    )
    expect(container).toMatchSnapshot()
  })
})
