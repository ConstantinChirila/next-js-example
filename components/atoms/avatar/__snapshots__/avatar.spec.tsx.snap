// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match a snapshot 1`] = `
.c0 {
  border-radius: 1.2rem;
  display: block;
  height: 6.4rem;
  object-fit: cover;
  object-position: center;
  width: 6.4rem;
}

<div>
  <img
    alt="hello alt"
    class="c0"
    src="http://placekitten.com/90/90"
  />
</div>
`;
